cmake/Platform/Mac/Install_mac.cmake

```diff
@@ -125,15 +128,14 @@ function(ly_post_install_steps)
             continue()
         endif()
         
-        install(SCRIPT ${CMAKE_BINARY_DIR}/runtime_install/$<CONFIG>/${target}.cmake COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME})
+        ly_install_run_script(${CMAKE_BINARY_DIR}/runtime_install/$<CONFIG>/${LY_BUILD_PERMUTATION}/${target}.cmake)
     endforeach()
 
-    install(CODE
-"ly_download_and_codesign_sdk_python()
-ly_codesign_sdk()"
-        COMPONENT ${CMAKE_INSTALL_DEFAULT_COMPONENT_NAME}
-    )
+    ly_install_run_code("
+        ly_download_and_codesign_sdk_python()
+        ly_codesign_sdk()
+        set(CMAKE_INSTALL_PREFIX ${LY_INSTALL_PATH_ORIGINAL})
+    ")
```

Comment:
Is this right?
Setting the `CMAKE_INSTALL_PREFIX` back to the original install prefix in the engine install step via this logic doesn't seem like it would work if the user invoked  the `cmake --install` option with the [--prefix](https://cmake.org/cmake/help/v3.21/manual/cmake.1.html#install-a-project) parameter.

I believe the prefix option would update the CMAKE_INSTALL_PREFIX *after* the LY_INSTALL_PATH_ORIGINAL would be set?

Reply:
Yes. I've been using `cmake --install` with the `--prefix` option for testing. The LY_INSTALL_PATH_ORIGINAL is set after CMAKE_INSTALL_PREFIX is set. Here's a snippet from the cmake_install.cmake file at the build root that CMake generates/invokes when we do cmake --install:

```
# Install script for directory: /Users/srikkant/O3DE/o3de

# Set the install prefix
if(NOT DEFINED CMAKE_INSTALL_PREFIX)
  set(CMAKE_INSTALL_PREFIX "/Users/srikkant/O3DE/o3de/install")
endif()
string(REGEX REPLACE "/$" "" CMAKE_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}")

# Set the install configuration name.
if(NOT DEFINED CMAKE_INSTALL_CONFIG_NAME)
  if(BUILD_TYPE)
    string(REGEX REPLACE "^[^A-Za-z0-9_]+" ""
           CMAKE_INSTALL_CONFIG_NAME "${BUILD_TYPE}")
  else()
    set(CMAKE_INSTALL_CONFIG_NAME "release")
  endif()
  message(STATUS "Install configuration: \"${CMAKE_INSTALL_CONFIG_NAME}\"")
endif()

# Set the component getting installed.
if(NOT CMAKE_INSTALL_COMPONENT)
  if(COMPONENT)
    message(STATUS "Install component: \"${COMPONENT}\"")
    set(CMAKE_INSTALL_COMPONENT "${COMPONENT}")
  else()
    set(CMAKE_INSTALL_COMPONENT)
  endif()
endif()

# Is this installation the result of a crosscompile?
if(NOT DEFINED CMAKE_CROSSCOMPILING)
  set(CMAKE_CROSSCOMPILING "FALSE")
endif()

# Set default install directory permissions.
if(NOT DEFINED CMAKE_OBJDUMP)
  set(CMAKE_OBJDUMP "/Applications/Xcode-beta.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/objdump")
endif()

set(CMAKE_BINARY_DIR "/Users/srikkant/O3DE/o3de/build/mac")

if("x${CMAKE_INSTALL_COMPONENT}x" STREQUAL "xCorex" OR NOT CMAKE_INSTALL_COMPONENT)
  include("/Users/srikkant/O3DE/o3de/build/mac/runtime_install/PreInstallSteps_mac.cmake")
endif()
```

So at the top of the file(beginning of install) it checks if CMAKE_INSTALL_PREFIX was passed in through command-line using the --prefix option. If it's not, it sets it to the default install prefix that O3DE specifies. The PreInstallSteps script is included after CMake's initial setup.

Reply:
Thanks for the clarification.